lib/presentation/features/details/bloc/details_bloc.dart

```diff
@@ -30,13 +31,13 @@ class DetailsBloc extends Bloc<DetailsEvent, DetailsState> {
           loading: false,
           character: characterViewData,
           series: seriesViewData,
-          error: null));
+          errorObject: null));
     } on DataRetrieveException {
-      emit(state.copyWith(loading: false, error: "slow internet connection"));
+      emit(state.copyWith(loading: false, errorObject: ErrorObject.slowInternetConnection()));
```

Comment:
Looks like you no need to use `errorObject` you can pass `DataRetrieveException error` directly in state like this:
```
} on DataRetrieveException catch (error) {
      emit(state.copyWith(loading: false, error: error));
```
and then on UI
```
String errorMessage;
if (state.error is DataRetrieveException) errorMessage = "..."
...
```

Reply:
done